src/Chapter3.hs

```diff
@@ -971,6 +1148,33 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Enum, Bounded, Eq, Show)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend x = case x of
+    Saturday -> True
+    Sunday -> True
+    _ -> False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay x = case x of
+    Sunday -> Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```haskell
    if x == maxBound then minBound ...
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂

Reply:
Sounds reasonable, but when I changed the code to this
```
nextDay :: DayOfWeek -> DayOfWeek
nextDay x = case x of
    maxBound -> minBound
    _ -> succ x
```
The compiler gives a warning "Pattern match is redundant". Although in fact it's not 🤔 If we remove the part `maxBound -> minBound`, then calling the function on Sunday will result in an error.
Is there a way to somehow tell the compiler that the warning should be suppressed?